rosidl_typesupport_opensplice_cpp/rosidl_typesupport_opensplice_cpp/rosidl_generator_dds_idl_extension.py

```diff
@@ -14,6 +14,21 @@
 
 from rosidl_generator_dds_idl import get_post_struct_lines as \
     get_default_post_struct_lines
+from rosidl_generator_dds_idl import idl_typename as \
+    default_idl_typename
+from rosidl_parser.definition import AbstractWString
+from rosidl_parser.definition import BasicType
+
+
+def idl_typename(type_):
+    typename = default_idl_typename(type_)
+    # OpenSplice doesn't support wchar / wstring, map to char / string instead
+    if isinstance(type_, BasicType) and type_.typename == 'wchar':
+        typename == 'char'
```

Comment:
A single `char` is smaller than `wchar`.DDS-XTypes 1.2 says `char` (via a couple hops to `CHAR_8_TYPE`) only holds 8 bits.

```
7.2.2.2.1.2.2 CHAR_8_TYPE
This specification does not define an encoding for the CHAR_8_TYPE . The only constraint is that it
shall be representable using 8 bits.
Rationale
By not specifying an encoding for CHAR_8_TYPE it is possible to use the 8-bit code-unit to either
store a single ISO-8859-1 character or alternatively a code-unit of a UTF-8 encoded string.
```

 `string/wstring` seems ok since it can become longer to hold more code points.  Maybe the `wchar` type should be changed to the 16 bit IDL type `unsigned short`?

Reply:
I am not sure it is necessary to emulate `wchar` for OpenSplice. If a user wants to use `wchar` with larger than 8-bit values they should simply not use OpenSplice. I don't think it makes sense for us to spend time on it.